File: Data Structures & Algorithms/string-encode-and-decode/submission-1.py
class Solution:

    def encode(self, strs: List[str]) -> str:
        self.lengths = [len(s) for s in strs]
        return ''.join(strs)

    def decode(self, s: str) -> List[str]:
        ans = []

        for i in self.lengths:
            ans.append(s[:i])
            s = s[i:]
        
        return ans

File: Data Structures & Algorithms/anagram-groups/submission-1.cpp
#include <vector>
#include <map>
#include <string>
#include <algorithm>
using namespace std;

class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        map<string, vector<string>> mpp;
        for(string s : strs){
            string std = s;
            sort(std.begin(), std.end());
            mpp[std].push_back(s);
        }
        vector<vector<string>> ans;
        for(auto p : mpp){
            ans.push_back(p.second);
        }
        return ans;
    }
};
